### DISCUSSION TITLE
how to add custom options to the webserver

### DISCUSSION THREAD

**Discussion by cesar23131** [easytarget/esp32-cam-webserver#310] [Q&A]
cant find anything about this how to i add a custom slider or toggle in the interface that executes a function in the .ino file

**Comment by cor-of-org** [comment]
I'm only commenting to see what anyone might say about this supremely stupid question.

**Reply by cesar23131** [reply]
sorry im new with the esp32 cam, maybe you can help me instead of trying to be funny by being an asshole, sounds good?

**Comment by cor-of-org** [comment]
Insult me then ask for help; nah, that's not working.

**Reply by cesar23131** [reply]
i didnt insult you, you insulted me first, im not here to  to start an argument, im just here for help, so if you dont want to help me, just stop, will you?

**Reply by easytarget** [reply]
I'm with @cesar23131 ; It's a very broad and hard to answer question, but not invalid.

**Comment by easytarget** [comment]
Hard to answer, and sorry I was not about to support you earlier. 

I think the best answer is to say: 'look at the code', this whole project has been about adding new controls and processing them in the main .ino code. If you compare the very first versions I publish with later ones you will see how I added controls. You have to add them to the html javascript and to the .ino. Adding javascript to add to the JSON status packets being sent by the web client, and adding code to the .ino to process that.

Anyway.. that said: Sorry; I am not doing any further development on this project. it is old code and somewhat obsolete. I am currently working on retiring it; I will try to link to a more actively updated fork when I do.
